component/web/oauth-common/src/main/java/org/gatein/security/oauth/openid/OpenIdProcessorImpl.java

```diff
@@ -426,34 +428,80 @@ protected JSONObject parseResponse(String httpResponse) throws JSONException {
 
         return uinfo;
     }
-    
+
     @Override
     public void revokeToken(OpenIdAccessTokenContext accessToken) throws OAuthException {
-    
+
     }
-    
+
     // Parse given String and add all scopes from it to given Set
     private void addScopesFromString(String scope, Set<String> scopes) {
         String[] scopes2 = scope.split(" ");
         for (String current : scopes2) {
             scopes.add(current);
         }
     }
-    
+
     protected URL sendAccessTokenRequest() throws IOException {
         if (log.isTraceEnabled())
             log.trace("AccessToken Request=" + this.accessTokenURL);
         return new URL(this.accessTokenURL);
     }
-    
+
     protected URL getTokenInfoURL() throws IOException {
         if (log.isTraceEnabled())
             log.trace("TokenInfo Request=" + this.tokenInfoURL);
         return new URL(this.tokenInfoURL);
     }
+
     protected URL getUserInfoURL() throws IOException {
         if (log.isTraceEnabled())
             log.trace("UserInfo Request=" + this.userInfoURL);
         return new URL(this.userInfoURL);
     }
-}
+
+    @Override
+    public void start() {
+        if (StringUtils.isBlank(this.wellKnownConfigurationUrl)) {
+            log.error("wellKnownConfigurationUrl is not configured");
+            return;
+        }
+        try {
+            String wellnownConfigurationContent = readUrl(this.wellKnownConfigurationUrl);
+            if (wellnownConfigurationContent != null) {
+                JSONObject json = new JSONObject(wellnownConfigurationContent);
+                this.authenticationURL = json.getString("authorization_endpoint");
+                this.accessTokenURL = json.getString("token_endpoint");
+                this.userInfoURL = json.getString("userinfo_endpoint");
+                this.tokenInfoURL = json.getString("token_endpoint");
+            }
+        } catch (JSONException e) {
+            log.error("Unable to read webKnownUrl content.", e.getMessage());
+        }
+    }
+
+    @Override
+    public void stop() {
+        // Nothing to stop
+    }
+
+    private static String readUrl(String urlString) {
```

Comment:
```suggestion
    private static String readUrl(URL url) {
```
This will allow you to fix the last sonar feedback